File: app/src/types/request/npc/bot.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { BotId } from '@/types/core/npc/bot';
import type { WorldId } from '@/types/core/world';

export interface GetBotCoordsRequest {
  readonly world: WorldId;
  readonly id: BotId;
}

export interface GetPublicBotRequest {
  readonly world: WorldId;
  readonly id: BotId;
}

export interface GetPublicBotsRequest {
  readonly world: WorldId;
}

File: crates/nil-payload/src/request/npc/bot.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use bon::Builder;
use nil_core::npc::bot::BotId;
use nil_core::world::config::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Builder, Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
#[cfg_attr(feature = "typescript", derive(ts_rs::TS))]
#[cfg_attr(feature = "typescript", ts(export))]
pub struct GetBotCoordsRequest {
  #[builder(start_fn, into)]
  pub world: WorldId,
  #[builder(into)]
  pub id: BotId,
}

#[derive(Builder, Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
#[cfg_attr(feature = "typescript", derive(ts_rs::TS))]
#[cfg_attr(feature = "typescript", ts(export))]
pub struct GetPublicBotRequest {
  #[builder(start_fn, into)]
  pub world: WorldId,
  #[builder(into)]
  pub id: BotId,
}

#[derive(Builder, Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
#[cfg_attr(feature = "typescript", derive(ts_rs::TS))]
#[cfg_attr(feature = "typescript", ts(export))]
pub struct GetPublicBotsRequest {
  #[builder(start_fn, into)]
  pub world: WorldId,
}
